core/templates/dev/head/components/fuzzy_rule_editor_directive.html

```diff
@@ -0,0 +1,2 @@
+<input type="hidden"/>
+<button type="button" class="btn btn-danger" ng-click="removeEntry()">Remove this input</button>
```

Comment:
Turns out that it works if you remove the "input type hidden" part and try clicking the button behind the select2. So maybe something is blocking something else, CSS wise...

I think it's also possible that the select2 stuff is messing up the DOM bindings somehow. Maybe you could think about how to have that be more separate. We already have a Select2DropdownDirective and perhaps you could use that as a component instead rather than rewriting it for this case.